assets/js/blocks/stock-filter/block.tsx

```diff
@@ -50,14 +54,11 @@ const StockStatusFilterBlock = ( {
 		false
 	);
 
-	const [ hideOutOfStockItems ] = useState(
-		getSetting( 'hideOutOfStockItems', false )
-	);
-	const [ { outofstock, ...otherStockStatusOptions } ] = useState(
-		getSetting( 'stockStatusOptions', {} )
-	);
+	const [ { outofstock, ...otherStockStatusOptions } ] = useState<
+		Record< string, string >
+	>( getSetting( 'stockStatusOptions', {} ) );
 	const [ STOCK_STATUS_OPTIONS ] = useState(
```

Comment:
Same here, why is `STOCK_STATUS_OPTIONS` stored in the state if it's never changed? :confused:

Reply:
Went through the history and found the reason for this in https://github.com/woocommerce/woocommerce-blocks/pull/4682

> After discussing with @tjcafferkey we decided that these usages of useMemo were "over-optimisations". The length of STOCK_STATUS_OPTIONS will usually be 3, so iterating over it is cheap, and not worth memoizing given the complexity this adds to the component.
> Because the value of initialOptions is used as a dependency to the useEffect further down this file, setting these values as internal component state seemed the best way forward to stop this effect running every render.

Reply:
Oh, thanks for digging on it and finding the source. For cases like this, isn't the recommended solution to use [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref)?

Reply:
@Aljullu Brilliant! Look like `useRef` is perfect for this case. I updated the code per your suggestion 🚀